core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationEditorDirective.js

```diff
@@ -0,0 +1,149 @@
+ */
+
+oppia.directive('stateTranslationEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_editor/translation_tab/' +
+        'state_translation_editor_directive.html'),
+      controller: [
+        '$scope', '$uibModal', 'EditabilityService', 'ExplorationStatesService',
+        'StateEditorService', 'StateWrittenTranslationsService',
+        'TranslationLanguageService', 'TranslationTabActiveContentIdService',
+        'WrittenTranslationObjectFactory', function(
+            $scope, $uibModal, EditabilityService, ExplorationStatesService,
+            StateEditorService, StateWrittenTranslationsService,
+            TranslationLanguageService, TranslationTabActiveContentIdService,
+            WrittenTranslationObjectFactory) {
+          $scope.HTML_SCHEMA = {
+            type: 'html',
+            ui_config: {
+              hide_complex_extensions: 'true'
+            }
+          };
+          var showMarkAllAudioAsNeedingUpdateModalIfRequired = function(
+              contentId) {
+            var stateName = StateEditorService.getActiveStateName();
+            var state = ExplorationStatesService.getState(stateName);
+            var contentIdsToAudioTranslations = (
+              state.contentIdsToAudioTranslations);
+            if (contentIdsToAudioTranslations.hasUnflaggedAudioTranslations(
+              contentId)) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/components/forms/' +
+                  'mark_all_audio_as_needing_update_modal_directive.html'),
+                backdrop: true,
+                resolve: {
+                  considerTranslation: function() {
+                    return false;
+                  }
+                },
+                controller: 'MarkAllAudioAsNeedingUpdateController'
+              }).result.then(function() {
+                contentIdsToAudioTranslations.markAllAudioAsNeedingUpdate(
+                  contentId);
+                ExplorationStatesService.saveContentIdsToAudioTranslations(
+                  stateName, contentIdsToAudioTranslations);
+              });
+            }
+          };
+          var contentId = null;
+          var langaugeCode = null;
+          $scope.activeWrittenTranslation = null;
+          $scope.translationEditorIsOpen = false;
+          $scope.isTranslatable = EditabilityService.isTranslatable;
+          $scope.$on('activeContentIdChanged', function() {
+            initEditor();
+          });
+          $scope.$on('activeLanguageChanged', function() {
+            initEditor();
+          });
+
+          var initEditor = function() {
+            $scope.activeWrittenTranslation = null;
+            $scope.translationEditorIsOpen = false;
```

Comment:
Referring to [the comment](https://docs.google.com/document/d/1AmPgTnrQLXQinPivF7eiwQ89sC3By-087atGKEYm1Uw/edit?disco=AAAACePreoU) in the design doc.. Shouldn't this conditionally be true/false? Instead of being always false.

Reply:
Thanks for the clarification!